buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-						h.Mkfile(t,
-							"[[unmet]]\n"+
-								`name = "unknown-dep"`+"\n",
-							filepath.Join(appDir, "build-A-v1.toml"),
-						)
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNotNil(t, err)
-						expected := "must match a requested dependency"
-						h.AssertStringContains(t, err.Error(), expected)
-					})
-				})
-			})
-
-			when("there is more than one default=true process", func() {
-				it.Before(func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-				})
-
-				when("the processes are with the same type", func() {
-					it("should error", func() {
-						h.Mkfile(t,
-							`[[processes]]`+"\n"+
-								`type = "some-type"`+"\n"+
-								`command = "some-cmd"`+"\n"+
-								`default = true`+"\n"+
-								`[[processes]]`+"\n"+
-								`type = "some-type"`+"\n"+
-								`command = "some-other-cmd"`+"\n"+
-								`default = true`+"\n",
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNotNil(t, err)
-						expected := "multiple default process types aren't allowed"
-						h.AssertStringContains(t, err.Error(), expected)
-					})
-				})
 
-				when("the processes are with different types", func() {
-					it("should error", func() {
-						h.Mkfile(t,
-							`[[processes]]`+"\n"+
-								`type = "some-type"`+"\n"+
-								`command = "some-cmd"`+"\n"+
-								`default = true`+"\n"+
-								`[[processes]]`+"\n"+
-								`type = "other-type"`+"\n"+
-								`command = "other-cmd"`+"\n"+
-								`default = true`+"\n",
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNotNil(t, err)
-						expected := "multiple default process types aren't allowed"
-						h.AssertStringContains(t, err.Error(), expected)
+						when("launch.toml", func() {
+							it("ignores process working directory and warns", func() {
+								h.Mkfile(t,
+									"[[processes]]\n"+
+										`working-dir = "/working-directory"`+"\n"+
+										`type = "some-type"`+"\n",
+									filepath.Join(appDir, "launch-A-v1.toml"),
+								)
+								br, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+								h.AssertNil(t, err)
+								h.AssertEq(t, len(br.Processes), 1)
+								h.AssertEq(t, br.Processes[0].WorkingDirectory, "")
+								assertLogEntry(t, logHandler, "Warning: process working directory isn't supported in this buildpack api version. Ignoring working directory for process 'some-type'")
+							})
+						})
 					})
 				})
 			})
-
-			when("the launch, cache and build flags are in the top level", func() {
-				it("should error", func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-					h.Mkdir(t,
-						filepath.Join(layersDir, "A"),
-						filepath.Join(appDir, "layers-A-v1", "layer"),
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L415-L429